tdrs-backend/tdpservice/scheduling/tasks.py

```diff
@@ -0,0 +1,112 @@
+           ):
+    """
+    Upload to SFTP server.
+
+    This task uploads the file in DataFile object with pk = data_file_pk
+    to sftp server as defined in Settings file
+    """
+    # Upload file
+    data_file = DataFile.objects.get(id=data_file_pk)
+    file_transfer_record = LegacyFileTransfer(
+        data_file=data_file,
+        uploaded_by=data_file.user,
+        file_name=data_file.create_filename(),
+    )
+
+    def create_dir(directory_name, sftp_server):
+        """Code snippet to create directory in SFTP server."""
+        try:
+            sftp_server.chdir(directory_name)  # Test if remote_path exists
+        except IOError:
+            sftp_server.mkdir(directory_name)  # Create remote_path
+            sftp_server.chdir(directory_name)
+
+    try:
+        # Create directory names for ACF titan
+        destination = str(data_file.create_filename())
+        today_date = datetime.datetime.today()
+        upper_directory_name = today_date.strftime('%Y%m%d')
+        lower_directory_name = today_date.strftime(str(data_file.year) + '-' + str(data_file.quarter))
+
+        # Paramiko need local file
+        f = data_file.file.read()
+        with open(destination, 'wb') as f1:
+            f1.write(f)
+            file_transfer_record.file_size = f1.tell()
+            file_transfer_record.file_shasum = hashlib.sha256(f).hexdigest()
+            f1.close()
+
+        # Paramiko SSH connection requires private key as file
+        temp_key_file = write_key_to_file(local_key)
+        os.chmod(temp_key_file, 0o600)
+
+        # Create SFTP/SSH connection
+        transport = paramiko.SSHClient()
+        transport.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        pkey = paramiko.RSAKey.from_private_key_file(temp_key_file)
+        transport.connect(server_address,
+                          pkey=pkey,
+                          username=username,
+                          port=port,
+                          look_for_keys=False,
+                          disabled_algorithms={'pubkeys': ['rsa-sha2-512', 'rsa-sha2-256']})
+        # remove temp key file
+        os.remove(temp_key_file)
+        sftp = transport.open_sftp()
+
+        # Create remote directory
+        create_dir(upper_directory_name, sftp_server=sftp)
+        create_dir(lower_directory_name, sftp_server=sftp)
+        create_dir(settings.ACFTITAN_DIECTORY, sftp_server=sftp)
+
+        # Put the file in SFTP server
+        sftp.put(destination, destination)
+
+        # Delete temp file
+        os.remove(destination)
+        logger.info('File {} has been successfully uploaded to {}'.format(destination, server_address))
+
+        # Add the log LegacyFileTransfer
+        file_transfer_record.result = LegacyFileTransfer.Result.COMPLETED
+        file_transfer_record.save()
+        transport.close()
+        return True
+
+    except Exception as e:
+        logger.error('Failed to upload {} with error:{}'.format(destination, e))
```

Comment:
As discussed with @raftmsohani, this large try/except has the downside that we won't have very granular errors stored in the model -- only this transient logging message. I don't really like the idea of storing an exception and/or stacktrace in the database -- that is something that should be reserved for logging. I think this is fine but potentially a future TODO if we need to revisit SFTP in the future. This entire feature is supposed to just be a stop-gap measure until ~v3.2 so I don't want further scope-creep on this ticket.